lang/cs/Org.Apache.REEF.Wake/Remote/Impl/DefaultRemoteManager.cs

```diff
@@ -142,22 +147,35 @@ public IObserver<T> GetRemoteObserver(IPEndPoint remoteEndpoint)
             }
 
             ProxyObserver remoteObserver;
-            if (!_cachedClients.TryGetValue(remoteEndpoint, out remoteObserver))
+            if (!_enableCaching)
             {
-                TransportClient<IRemoteEvent<T>> client = 
-                    new TransportClient<IRemoteEvent<T>>(remoteEndpoint, _codec, _observerContainer);
-                var msg = string.Format("NewClientConnection: Local {0} connected to Remote {1}",
-                    client.Link.LocalEndpoint.ToString(), 
-                    client.Link.RemoteEndpoint.ToString());
-                LOGGER.Log(Level.Info, msg);
-
-                remoteObserver = new ProxyObserver(client);
+                remoteObserver = CreateRemoteObserver(remoteEndpoint);
+            }
+            else if (!_cachedClients.TryGetValue(remoteEndpoint, out remoteObserver))
+            {
+                remoteObserver = CreateRemoteObserver(remoteEndpoint);
                 _cachedClients[remoteEndpoint] = remoteObserver;
             }
 
             return remoteObserver;
         }
 
+        private ProxyObserver CreateRemoteObserver(IPEndPoint remoteEndpoint)
```

Comment:
You can cast it to IDisposable.

-----Original Message-----
From: "Andrew Chung" <notifications@github.com>
Sent: ‎3/‎28/‎2016 1:46 PM
To: "apache/reef" <reef@noreply.github.com>
Cc: "Boris Shulman" <shulmanb@gmail.com>
Subject: Re: [apache/reef] [REEF-1142] Making caching configurable in theremote managers (#905)

In lang/cs/Org.Apache.REEF.Wake/Remote/Impl/DefaultRemoteManager.cs:
>                  _cachedClients[remoteEndpoint] = remoteObserver;
>              }
>  
>              return remoteObserver;
>          }
>  
> +        private ProxyObserver CreateRemoteObserver(IPEndPoint remoteEndpoint)
How would you close the connections? The returned type, IObserver<T>, is not IDisposable.
—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub